### PR TITLE
Fix #766 : Replace profile activity to profile chooser activity where needed

### LINKED ISSUE #766: Replace ProfileActivity to ProfileChooserActivity where needed
Currently, the activity in which user chooses the profile is named as `ProfileActivity` in code which is not appropriate according to the functionality of that activity. So, all the files corrensponding to choosing user profiles need to be changed to `ProfileChooserActivity`

Changes that are need to done: 

Rename `ProfileActivity` to `ProfileChooserActivity`
Rename `ProfileActivityPresenter` to `ProfileChooserActivityPresenter`
Rename `profile_activity.xml` to `profile_chooser_activity.xml`

Also check the test files too.

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes #766 - Replace profileActivity with ProfileChooserActivity where needed
Rename ProfileActivity to ProfileChooserActivity
Rename ProfileActivityPresenter to ProfileChooserActivityPresenter
Rename profile_activity.xml to profile_chooser_activity.xml

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.
Reviewers - @rt4914 @anandwana001

### CODE REVIEW

**Comment by prayutsu** [comment]
@rt4914 Successfully renamed the three files

**Comment by prayutsu** [comment]
@anandwana001 Hey can you help me identify the errors in the third test. Where I am lagging?

**Comment by anandwana001** [comment]
> @anandwana001 Hey can you help me identify the errors in the third test. Where I am lagging?

Sure, let me have a look on this

**Comment by anandwana001** [comment]
As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.

**Comment by prayutsu** [comment]
> As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.

  private fun setUpTestApplicationComponent() {
    DaggerAddProfileActivityTest_TestApplicationComponent.builder()
      .setApplication(ApplicationProvider.getApplicationContext())
      .build()
      .inject(this)
  }

  private fun setUpTestApplicationComponent() {
    DaggerAddProfileChooserActivityTest_TestApplicationComponent.builder()
      .setApplication(ApplicationProvider.getApplicationContext())
      .build()
      .inject(this)
  }

@anandwana001 I have tried changing "DaggerAddProfileActivityTest_TestApplicationComponent" to DaggerAddProfileChooserActivity_TestApplicationComponent". But it is still showing error.

**Comment by anandwana001** [comment]
> > As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.
> 
>   private fun setUpTestApplicationComponent() {
>     DaggerAddProfileActivityTest_TestApplicationComponent.builder()
>       .setApplication(ApplicationProvider.getApplicationContext())
>       .build()
>       .inject(this)
>   }
> 
>   private fun setUpTestApplicationComponent() {
>     DaggerAddProfileChooserActivityTest_TestApplicationComponent.builder()
>       .setApplication(ApplicationProvider.getApplicationContext())
>       .build()
>       .inject(this)
>   }
> 
> @anandwana001 I have tried changing "DaggerAddProfileActivityTest_TestApplicationComponent" to DaggerAddProfileChooserActivity_TestApplicationComponent". But it is still showing error.
> 

Have you tried running test cases in this file? Are they failing?

**Comment by prayutsu** [comment]
> > > As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.
> > 
> > 
> > private fun setUpTestApplicationComponent() {
> > DaggerAddProfileActivityTest_TestApplicationComponent.builder()
> > .setApplication(ApplicationProvider.getApplicationContext())
> > .build()
> > .inject(this)
> > }
> > private fun setUpTestApplicationComponent() {
> > DaggerAddProfileChooserActivityTest_TestApplicationComponent.builder()
> > .setApplication(ApplicationProvider.getApplicationContext())
> > .build()
> > .inject(this)
> > }
> > @anandwana001 I have tried changing "DaggerAddProfileActivityTest_TestApplicationComponent" to DaggerAddProfileChooserActivity_TestApplicationComponent". But it is still showing error.
> 
> Have you tried running test cases in this file? Are they failing?

No, Let me try then.
I simply need to click on the green play buttons (near the line numbers) to test them?

**Comment by anandwana001** [comment]
> > > > As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.
> > > 
> > > 
> > > private fun setUpTestApplicationComponent() {
> > > DaggerAddProfileActivityTest_TestApplicationComponent.builder()
> > > .setApplication(ApplicationProvider.getApplicationContext())
> > > .build()
> > > .inject(this)
> > > }
> > > private fun setUpTestApplicationComponent() {
> > > DaggerAddProfileChooserActivityTest_TestApplicationComponent.builder()
> > > .setApplication(ApplicationProvider.getApplicationContext())
> > > .build()
> > > .inject(this)
> > > }
> > > @anandwana001 I have tried changing "DaggerAddProfileActivityTest_TestApplicationComponent" to DaggerAddProfileChooserActivity_TestApplicationComponent". But it is still showing error.
> > 
> > Have you tried running test cases in this file? Are they failing?
> 
> No, Let me try then.
> I simply need to click on the green play buttons (near the line numbers) to test them?

Yes, you can run test cases individually as well as you can run all at once by clicking over gree play icon near class at top of the file where class name is mentioned. 
Also, you can see the difference in the menu bar near play icon, the configuration will get mentioned there what sort of thing the IDE is running.

**Comment by prayutsu** [comment]
> > > > > As we are on updating the names in this PR, there is a test file `AddProfileChooserActivityTest` after updating the name, but inside it where we are setting up the dagger the name is not updated.
> > > > 
> > > > 
> > > > private fun setUpTestApplicationComponent() {
> > > > DaggerAddProfileActivityTest_TestApplicationComponent.builder()
> > > > .setApplication(ApplicationProvider.getApplicationContext())
> > > > .build()
> > > > .inject(this)
> > > > }
> > > > private fun setUpTestApplicationComponent() {
> > > > DaggerAddProfileChooserActivityTest_TestApplicationComponent.builder()
> > > > .setApplication(ApplicationProvider.getApplicationContext())
> > > > .build()
> > > > .inject(this)
> > > > }
> > > > @anandwana001 I have tried changing "DaggerAddProfileActivityTest_TestApplicationComponent" to DaggerAddProfileChooserActivity_TestApplicationComponent". But it is still showing error.
> > > 
> > > 
> > > Have you tried running test cases in this file? Are they failing?
> > 
> > 
> > No, Let me try then.
> > I simply need to click on the green play buttons (near the line numbers) to test them?
> 
> Yes, you can run test cases individually as well as you can run all at once by clicking over gree play icon near class at top of the file where class name is mentioned.
> Also, you can see the difference in the menu bar near play icon, the configuration will get mentioned there what sort of thing the IDE is running.
@anandwana001 yes they are failing, this is the error -
Unresolved reference: DaggerAddProfileActivityTest_TestApplicationComponent

**Comment by prayutsu** [comment]
@anandwana001 Now I am renaming it.

**Comment by prayutsu** [comment]
> @anandwana001 Now I am renaming it.

23 out of 48 tests failed after renaming

**Comment by prayutsu** [comment]
@anandwana001 @rt4914 PTAL

File: app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt [hunk 1]
```diff
@@ -254,7 +254,7 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
             }
             .setPositiveButton(R.string.home_activity_back_dialog_exit) { _, _ ->
               // TODO(#322): Need to start intent for ProfileActivity to get update. Change to finish when live data bug is fixed.
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt]: Update this comment too.

File: app/src/main/java/org/oppia/app/home/HomeActivity.kt [hunk 2]
```diff
@@ -44,7 +44,7 @@ class HomeActivity : InjectableAppCompatActivity(), RouteToTopicListener {
       }
       .setPositiveButton(R.string.home_activity_back_dialog_exit) { _, _ ->
         // TODO(#322): Need to start intent for ProfileActivity to get update. Change to finish when live data bug is fixed.
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/HomeActivity.kt]: Update this comment too.

File: app/src/main/java/org/oppia/app/profile/AddProfileActivity.kt [hunk 3]
```diff
@@ -29,7 +29,7 @@ class AddProfileActivity : InjectableAppCompatActivity() {
 
   override fun onSupportNavigateUp(): Boolean {
     // TODO(#322): Need to start intent for ProfileActivity to get update. Change to finish when live data bug is fixed.
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/AddProfileActivity.kt]: Update this comment too.

File: app/src/main/java/org/oppia/app/profile/ProfileChooserActivity.kt [hunk 4]
```diff
@@ -7,13 +7,13 @@ import org.oppia.app.activity.InjectableAppCompatActivity
...
 
 /** Activity that controls profile creation and selection. */
-class ProfileActivity : InjectableAppCompatActivity() {
+class ProfileChooserActivity : InjectableAppCompatActivity() {
   @Inject
-  lateinit var profileActivityPresenter: ProfileActivityPresenter
+  lateinit var profileChooserActivityPresenter: ProfileChooserActivityPresenter
 
   companion object {
     fun createProfileActivity(context: Context): Intent {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/ProfileChooserActivity.kt]: Update this function too.

File: app/src/sharedTest/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityTest.kt [hunk 5]
```diff
@@ -458,7 +472,7 @@ class AdministratorControlsActivityTest {
       onView(withText(R.string.log_out_dialog_message)).inRoot(isDialog())
         .check(matches(isDisplayed()))
       onView(withText(R.string.log_out_dialog_okay_button)).perform(click())
-      intended(hasComponent(ProfileActivity::class.java.name))
+      intended(hasComponent(ProfileChooserActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/administratorcontrols/AdministratorControlsActivityTest.kt:475]: Update the test name too.

File: app/src/sharedTest/java/org/oppia/app/profile/AddProfileChooserActivityTest.kt [hunk 6]
```diff
@@ -59,7 +59,7 @@ import javax.inject.Singleton
 
 @RunWith(AndroidJUnit4::class)
 @LooperMode(LooperMode.Mode.PAUSED)
-class AddProfileActivityTest {
+class AddProfileChooserActivityTest {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/profile/AddProfileChooserActivityTest.kt:62]: Update all test cases names too.

File: app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt [hunk 7]
```diff
@@ -634,7 +634,7 @@ class ProfileChooserFragmentTest {
   }
 
   private fun createProfileActivityIntent(): Intent {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt]: Update function name

File: app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt [hunk 8]
```diff
@@ -125,7 +125,7 @@ class SplashActivityTest {
     activityTestRule.launchActivity(null)
     testCoroutineDispatchers.advanceUntilIdle()
 
-    intended(hasComponent(ProfileActivity::class.java.name))
+    intended(hasComponent(ProfileChooserActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:128]: Update testcase name

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt [hunk 9]
```diff
@@ -351,7 +351,7 @@ class NavigationDrawerTestActivityTest {
       onView(withText(R.string.menu_switch_profile)).perform(click())
       onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))
       onView(withText(R.string.home_activity_back_dialog_exit)).perform(click())
-      intended(hasComponent(ProfileActivity::class.java.name))
+      intended(hasComponent(ProfileChooserActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:354]: Update test case name

**Review by rt4914 — COMMENTED** [review]
Thanks @prayutsu 

Suggested changes. Also, please make sure that your branch name uses all `small characters` separated by `underscore`.

**Comment by prayutsu** [comment]
@rt4914 Closing this PR as a new one has been created #1700

File: app/src/sharedTest/java/org/oppia/app/profile/AddProfileChooserActivityTest.kt (hunk 6 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/app/profile/AddProfileChooserActivityTest.kt:62]: @rt4914 I need to revert all the changes in the file?

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/profile/AddProfileChooserActivityTest.kt:62]: Yes, I missed this in this PR but I have commented accordingly in other PR.

**Pull request closed without merging**